### PR TITLE
Working with gridbag layout in Java

### LINKED ISSUE #4620: Working with GridBag layout in Java
#### Proposed title of article
Working with GridBag Layout in Java

#### Proposed article introduction
Migrating from building console apps to GUI apps is one of the best fantasies of every budding developer. But the spoiler comes when the GUI isn’t pleasing to the sight.
Layouts are one of the most important parts of building GUI as it determines the looks and feels of our apps. Knowing the best layout manager to use that fits your need is as important as using the layout itself. GridBag layout manager is used to build complicated panels in our Java apps. This is made possible by some of the features it provides which include specifying rows and columns, spans, alignment, etc.
In this tutorial, we will look at the different types of layout managers in Java. And then focus on exploring the GridBag layout manager. We will walk from exploring the methods in the layout, how to use the constraints, and then build a project with the knowledge gained in the tutorial.

#### Key takeaways 
* Understand the basics of layout in Java
* Differentiate between GridBag and other layout managers in Java
* Understand the core concepts of GridBag layout manager
* Know the different approaches to GridBag layout manager in Java
* Know when to use GridBag layout manager as well as the pros and cons
* Build Java apps using GridBag layout manager

#### Article Quality
This article explores some layout managers in Java and then focuses on GridBag throughout the remaining part. It is beginner-friendly and provides an in-depth knowledge of GridBag layout manager by using both screenshots and code snippets to explain all of the components of a GridBag layout, method of usage, GridBag constraints, and many more. 

#### References
N/A

### PR DESCRIPTION
**Pull request by princeibs**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This fixes #4620

### CODE REVIEW

**Comment by paulodhiambo** [comment]
@princeibs please link this article to an issue.

**Comment by princeibs** [comment]
This article closes issue #4620

@paulodhiambo

**Comment by ahmadmardeni1** [comment]
Please always use the template provided by us.
I edited your initial comment and added the template. Check what matches your article.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/working-with-gridbag-layout-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+* Border layout
+* Box layout
+* Grid layout
+* Group layout
+* GridBag layout
+
+In this article, we will be considering the Gridbag layout and building an amazing user interface with it.
+
+### What is GridBag Layout Manager
+The gridbag layout manager places components in rows and columns similar to the grid layout. But with gridbag layout, you can specify the size of each row or column. You can also decide whether a component spans many rows or columns. Gridbag layout is one of the most flexible and complex layout manager among all that Java provides. It is used to build complicated UIs and it is one of the most used layout managers. When working with components of different sizes, then the gridbag layout is for you.
```
Comment (paulodhiambo) [inline content/articles/working-with-gridbag-layout-in-java/index.md:58]: Be consistent, rename all occurrence of the word "gridbag" to "Gridbag"

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
Hello @princeibs please always ensure you are consistent with your naming, I'll proceed with tech review.

**Comment by princeibs** [comment]
I have updated the article to reflect the requested change. Thanks
@paulodhiambo

**Comment by paulodhiambo** [comment]
@princeibs why is your PR having more than 600 filea

**Comment by paulodhiambo** [comment]
@princeibs please sync your forked repo with the section repository and create another PR. I'll close this PR. For any assistance, ping me through slack.

**Pull request closed without merging**